v0/destinations/kafka/transform.js

```diff
@@ -18,19 +19,39 @@ function batch(destEvents) {
 }
 
 function process(event) {
-  const integrationsObj = getIntegrationsObj(event.message, "kafka");
-  const { schemaId } = integrationsObj || {};
+  const { message } = event;
```

Comment:
Post passing the destConfig in event, we can have the entire handling here:
If conf.MultiTopicSupport=false, take destConf.Topic
Else, getIntegrationsObj/event mapping based to identify the topic & fallback to default destConf.Topic